src/plugins/data/public/search/errors/painless_error.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiButton, EuiSpacer, EuiText, EuiCodeBlock } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { ApplicationStart } from 'kibana/public';
+import { KbnError } from '../../../../kibana_utils/common';
+import { EsError, isEsError } from './types';
+import { IEsSearchRequest } from '..';
+
+export class PainlessError extends KbnError {
+  painlessStack?: string;
+  constructor(err: EsError, request: IEsSearchRequest) {
+    const rootCause = getRootCause(err as EsError);
+
+    super(
+      i18n.translate('data.painlessError.painlessScriptedFieldErrorMessage', {
+        defaultMessage: "Error with Painless scripted field '{script}'.",
+        values: { script: rootCause?.script },
+      })
+    );
+    this.painlessStack = rootCause?.script_stack ? rootCause?.script_stack.join('\n') : undefined;
+  }
+
+  public getErrorMessage(application: ApplicationStart) {
+    function onClick() {
+      application.navigateToApp('management', {
+        path: `/kibana/indexPatterns`,
+      });
+    }
+
+    return (
+      <>
+        {this.message}
+        <EuiSpacer size="s" />
+        <EuiSpacer size="s" />
+        {this.painlessStack ? (
+          <EuiCodeBlock data-test-subj="painlessStackTrace" isCopyable={true} paddingSize="s">
+            {this.painlessStack}
+          </EuiCodeBlock>
+        ) : null}
+        <EuiText textAlign="right">
+          <EuiButton color="danger" onClick={onClick} size="s">
+            <FormattedMessage id="data.painlessError.buttonTxt" defaultMessage="Edit script" />
+          </EuiButton>
+        </EuiText>
+      </>
```

Comment:
This is nice, will help users debugging error 👍! 
I think you're using the code of the scripted field and not the name of the field in the error message?
Because the name of the field was: `invalid_scripted_field`

Furthermore I think it would be useful to also provide a way to "See the full error"
<img width="399" alt="Bildschirmfoto 2020-09-23 um 11 29 16" src="https://user-images.githubusercontent.com/463851/93994679-70ee3a00-fd90-11ea-9a63-a7108d65cd22.png">

Reply:
Unfortunately there's no good way to have two actions on a toast. The design team discourages this too.
As this is not the main focus of this PR, can I leave it for you to discuss with the design team?

Reply:
sure! this is already a nice improvement!

Reply:
So is it approved :-P @kertal

Reply:
did you check about the error message? `Error with Painless scripted field 'invalid'`. I think invalid is not the name of the field, but the code.

Reply:
@kertal, the response surpridingly does not contain the field name.
I copied this logic from the old painless error.

I improved the message so it makes more sense.

Reply:
thx!